### PR TITLE
Add JS notifications and functionality to docs

### PR DESCRIPTION
**Pull request by gozalig**
## Done

Added descriptions of JS functionality, added some warnings about JS being needed.

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/
- Check the code and website run ok.

### CODE REVIEW

**Comment by anthonydillon** [comment]
I am off tomorrow and will be AFK. Could someone review this branch please.

**Comment by kwm14** [comment]
## Design review
@gozalig looking good but just noticed double notification on **patterns/list-tree/**

Screenshot 2019-08-16 at 09 53 13

File: docs/utilities/baseline-grid.md [hunk 1]
```diff
@@ -12,3 +12,9 @@ You can apply this utility to an element (such as `<body>`) to give it a series
     class="js-example">
 View example of the baseline grid utility
 </a>
+
+<div class="p-notification--caution">
+  <p class="p-notification__response">
+    <span class="p-notification__status">Warning:</span>This pattern requires JS to be functional.
+  </p>
+</div>
```
Comment (anthonydillon) [inline docs/utilities/baseline-grid.md]: Not required here as the grid lines are not part of the framework itself.

File: docs/patterns/notification.md [hunk 2]
```diff
@@ -19,6 +19,12 @@ The default variant should be used to display global messages.
 View example of the default notification pattern
 </a>
 
+<div class="p-notification--caution">
+  <p class="p-notification__response">
+    <span class="p-notification__status">Warning:</span>This pattern requires JS to be functional.
+  </p>
+</div>
```
Comment (anthonydillon) [inline docs/patterns/notification.md]: Could you move this to the notification pattern that requires JS. I assume it's the one that can be hidden.

File: docs/patterns/accordion.md [hunk 3]
```diff
@@ -23,6 +23,16 @@ Each tab styling can be changed to open or collapse using `aria-expanded`, set `
...
 
+<div class="p-notification--caution">
+  <p class="p-notification__response">
+    <span class="p-notification__status">Warning:</span>This pattern requires JS to be functional.
+  </p>
+</div>
+
+### Functionality
+
+Use toggle `aria-expanded` to reveal or conceal the content of an accordion.
```
[the remark this answers is not in the document]
Comment (kwm14) [reply docs/patterns/accordion.md]: Updated 👍

File: docs/patterns/list-tree.md [hunk 4]
```diff
@@ -15,6 +15,18 @@ Each directory can be opened or collapse using `aria-hidden`, set `true` for ope
...
+  <p class="p-notification__response">
+    <span class="p-notification__status">Warning:</span>This pattern requires JS to be functional.
+  </p>
+</div>
+
+<div class="p-notification--caution">
+  <p class="p-notification__response">
+    <span class="p-notification__status">Warning:</span>This pattern requires JS to be functional.
+  </p>
+</div>
```
[the remark this answers is not in the document]
Comment (kwm14) [reply docs/patterns/list-tree.md]: Removed duplicate 👍

File: docs/settings/spacing-settings.md [hunk 5]
```diff
@@ -17,6 +17,12 @@ Vanilla uses a default spacing unit of `.5rem` (`8px`) as a basis to calculate s
 View example of the baseline grid utility
 </a>
 
+<div class="p-notification--caution">
+  <p class="p-notification__response">
+    <span class="p-notification__status">Warning:</span>This pattern requires JS to be functional.
+  </p>
+</div>
```
[the remark this answers is not in the document]
Comment (kwm14) [reply docs/settings/spacing-settings.md]: Removed 👍

File: docs/utilities/baseline-grid.md (hunk 1 above)
Comment (kwm14) [reply docs/utilities/baseline-grid.md]: Removed 👍

File: docs/patterns/notification.md (hunk 2 above)
Comment (kwm14) [reply docs/patterns/notification.md]: Updated 👍

File: docs/patterns/table-of-contents.md [hunk 6]
```diff
@@ -13,6 +13,12 @@ A table of contents can be used to display supplementary links to a page.
 View example of the table of contents pattern
 </a>
 
+<div class="p-notification--caution">
```
[the remark this answers is not in the document]
Comment (kwm14) [reply docs/patterns/table-of-contents.md]: @anthonydillon I can't find sorting under TOC? Do you mean https://docs.vanillaframework.io/base/tables/#sortable

**Comment by kwm14** [comment]
## Percy Review
- Wowza, increase font-size alert
- Will look into and fix up

4b3afe170ea4d9f75f3cb9fe1ca5aad6e66660a7d4117d034b99549b6a2b7dad

File: docs/patterns/accordion.md [hunk 7]
```diff
@@ -18,17 +18,21 @@ Each tab styling can be changed to open or collapse using `aria-expanded`, set `
...
 </div>
 
+<a href="/examples/patterns/accordion/"
+    class="js-example">
+View example of the accordion pattern
+</a>
+
+### Functionality
+
+Toggle `aria-expanded` to reveal or conceal the content of an accordion.
```
Comment (anthonydillon) [inline docs/patterns/accordion.md]: Requires more to be fully functional:
"Please ensure the `aria-control` attribute matches an ID of an element. If `aria-expanded` is true, then the accordion will be open by default. When clicking on the `accordion__tab`, you must toggle the `aria-expanded` attribute on the toggle and the `aria-hidden` attribute on the panel."

File: docs/patterns/modal.md [hunk 8]
```diff
@@ -13,6 +13,12 @@ The modal component can be used to overlay an area of the screen which can conta
 View example of the modal pattern
 </a>
 
+<div class="p-notification--caution">
+  <p class="p-notification__response">
+    <span class="p-notification__status">Warning:</span>This pattern requires JS to be functional.
+  </p>
+</div>
+
```
Comment (anthonydillon) [inline docs/patterns/modal.md]: This pattern is missing information on how to close the model. We should explain what the `showModal` function is doing.

File: docs/patterns/navigation.md [hunk 9]
```diff
@@ -45,6 +45,12 @@ or more `p-subnav` components.
 View example of the sub-navigation pattern
 </a>
 
+<div class="p-notification--caution">
+  <p class="p-notification__response">
+    <span class="p-notification__status">Warning:</span>This pattern requires JS to be functional.
+  </p>
+</div>
```
Comment (anthonydillon) [inline docs/patterns/navigation.md]: Again we should explain the attributes which need to be toggled on click

**Comment by kwm14** [comment]
> A few missing explanations on the patterns. Also could we be consistent with the position of the JS warning and section around the pattern?

Sure, will add above the code block rather than below 👍

File: docs/patterns/accordion.md (hunk 7 above)
Comment (kwm14) [reply docs/patterns/accordion.md]: Updated 👍

File: docs/patterns/modal.md (hunk 8 above)
Comment (kwm14) [reply docs/patterns/modal.md]: Added 👍

**Comment by kwm14** [comment]
Comment addresses @anthonydillon so ready for re-review 💯

File: docs/patterns/navigation.md (hunk 9 above)
Comment (kwm14) [reply docs/patterns/navigation.md]: Added 👍

**Pull request merged**